doc/dev/kdb-record.md

```diff
@@ -0,0 +1,110 @@
+# `kdb record` Command
+
+The `kdb record` tooling enables the user to record configuration sessions. It also allows exporting and importing the configuration changes.
+
+Recording sessions are stored on a per-user basis. This implies that a recording session can only track changes made by a single user.
+
+## Session handling
+
+* `kdb record start [session_name]`: Starts a new recording session. If no session name is provided, a new random name is generated. If a session with the provided name already exists, changes will be appended.
+* `kdb record stop`: Stops the current recording session. 
+* `kdb record delete [session_name]`: Deletes the specified recording session. If no session name is provided, the current session will be deleted. Deleting a session will not stop the recording. 
+* `kdb record list-sessions`: Lists all sessions by their name
+* `kdb record changes [session_name]`: Shows all changes for the specified session. If no session name is provided, the current session will be used. 
+
+## Exporting configuration changes
+
+`kdb record export [session_name] [--format <format>]`: Will dump the changed keys of the specified session onto `stdout`. If no session name is provided, the current session will be used. 
```

Comment:
You should probably provide an example before this, since the current order makes it sound like `kdb record` doesn't actually record all changes. Until you get to the example, it sounds like you just get a diff between start and stop states.